src/types.ts

```diff
@@ -80,3 +77,14 @@ export interface ModalClassNames {
   walletDescription?: string
   textContent?: string
 }
+
+export enum State {
+  Initializing,
+  AttemptingAutoConnection,
+  // Don't call connect until this state is reached.
+  ReadyForConnection,
+  Connecting,
+  Connected,
+  Resetting,
```

Comment:
what's this one for?

Reply:
Sometimes the modals get stuck or the process just generally messes up somewhere, and the modals display a reset button to reset state and start the connection process over. It's more of an internal status, but I think it belongs here because it is mutually exclusive with the rest of the statuses.